src/protocols/gamma_control.rs

```diff
@@ -205,16 +193,16 @@ where
         _resource: &ZwlrGammaControlV1,
         data: &GammaControlState,
     ) {
-        if data.previous_gamma_ramp.is_none() {
+        if data.failed {
             return;
         }
         let ramp = data.previous_gamma_ramp.as_ref().unwrap();
-        state
-            .set_gamma(&data.output, ramp.to_vec())
-            .unwrap_or_else({
-                error!("Failed to reset gamma ramp");
-                |_| ()
-            });
+
+        if let Err(err) = state.set_gamma(&data.output, ramp.to_vec(), data.gamma_size.unwrap()) {
+            warn!("error resetting gamma: {err:?}");
+            return;
```

Comment:
We still need to call `state.destroy` in this case. Nothing we can really do about the fact that resetting gamma failed.